src/ui/components/DeckBuilder/types.ts

```diff
@@ -1,5 +1,6 @@
 import { ICard } from '../../../game/types'
 
 export interface DeckBuilderProps {
-  onDone: (deck: Map<ICard, number>) => void
+  onDone: (deck: Map<ICard, number>) => void | Promise<void>
```

Comment:
@jules I don't see a reason why this should ever return a synchronous `void`. Update this to be:

```suggestion
  onDone: (deck: Map<ICard, number>) => Promise<void>
```

And update any consumers of `DeckBuilder` as necessary.